ULC/Repl.hs

```diff
@@ -13,52 +13,61 @@ This is the top level interface between the user and the interpreter.
 module Repl where
 
 -- ULC Imports.
-import ULC         
+import ULC
   (Term(Var), substitute, vars, reductions, reduce)
-import Parser 
+import Parser
   (apply, term, pLet, pTerm, (+++))
 
 
 -- Tool Imports.
-import System.IO             (hFlush, stdout)
-import qualified Data.Map.Lazy as M 
+import qualified Data.Map.Lazy as M
 
+-- Haskeline imports
+import Control.Monad.Trans
+import Control.Monad.Trans.State.Strict
+import System.Console.Haskeline
 
 -- | Top-level repl function
 replMain :: IO ()
 replMain = do
   putStrLn "Welcome to the Untyped \x03bb-calculus REPL"
   putStrLn "Type some terms or press Enter to leave."
-  repl M.empty
+  evalStateT (runInputT settings repl) M.empty
 
+settings :: MonadException m => Settings m
```

Comment:
Please add a quick Haddock comment here, and ensure haddock has 100% coverage for the code :)